Импортируем библиотеку статистического анализа в Python

In [1]:
import scipy.stats as st

Введём наши данные результатов тестирования по трём видам бросков сначала до проведения методики, а затем после для **8 "A"** класса:

In [2]:
_8a_niz_do = [4, 5, 2, 3, 4, 6, 4, 3, 5, 7, 5, 4, 5, 2, 0, 4, 5, 6, 4, 7, 3, 4, 2, 5, 1, 4]
_8a_1ha_do = [5, 3, 4, 5, 5, 3, 3, 2, 5, 8, 4, 3, 6, 3, 2, 3, 4, 5, 6, 5, 5, 4, 3, 4, 3, 2]
_8a_gru_do = [4, 4, 5, 4, 4, 5, 5, 4, 3, 5, 6, 5, 5, 3, 2, 3, 4, 4, 4, 5, 5, 3, 3, 4, 4, 2]

_8a_niz_ps = [5, 5, 3, 4, 5, 6, 5, 4, 5, 7, 6, 4, 5, 4, 2, 4, 5, 6, 4, 7, 4, 4, 2, 5, 2, 4]
_8a_1ha_ps = [5, 5, 6, 5, 6, 5, 4, 5, 5, 8, 4, 5, 6, 5, 2, 3, 5, 6, 6, 5, 6, 5, 4, 5, 3, 4]
_8a_gru_ps = [4, 5, 5, 6, 5, 5, 5, 4, 4, 6, 6, 5, 6, 5, 4, 3, 5, 6, 4, 5, 6, 4, 4, 5, 4, 5]

Введём такие же данные теперь для **8 "Б"**:

In [3]:
_8b_niz_do = [3, 4, 3, 4, 4, 5, 5, 3, 2, 3, 4, 4, 3, 5, 3, 0, 4, 3, 2, 6, 4, 3, 5, 6]
_8b_1ha_do = [3, 4, 4, 5, 5, 4, 4, 5, 4, 5, 4, 4, 5, 4, 3, 2, 3, 4, 3, 4, 4, 4, 5, 6]
_8b_gru_do = [4, 5, 4, 4, 5, 3, 5, 5, 4, 5, 4, 3, 4, 5, 4, 3, 5, 5, 4, 4, 5, 4, 3, 5]

_8b_niz_ps = [3, 4, 4, 4, 4, 4, 6, 4, 2, 3, 5, 4, 4, 5, 5, 1, 3, 4, 2, 5, 4, 5, 5, 5]
_8b_1ha_ps = [2, 4, 4, 5, 5, 5, 4, 5, 5, 5, 4, 5, 6, 5, 3, 2, 3, 2, 3, 5, 3, 5, 5, 5]
_8b_gru_ps = [3, 4, 4, 4, 5, 4, 5, 4, 4, 5, 4, 4, 5, 5, 4, 3, 5, 5, 4, 4, 4, 4, 3, 5]

С помощью критерия **Манна-Уитни** найдём есть ли статистически значимая разница при бросках снизу между учениками из 8 "A" и 8 "Б" классов до проведения методики: 

In [13]:
st.mannwhitneyu(_8a_niz_do, _8b_niz_do)

MannwhitneyuResult(statistic=265.5, pvalue=0.1801046041972389)

Значение **pvalue = 0.1801.. больше уровня статистической значимости β = 0.05**.<br>
Делаем вывод, что статистически значимых различий при бросках снизу между учениками 8 "A" и 8 "Б" до проведения методики не обнаружено.


In [5]:
st.mannwhitneyu(_8a_1ha_do, _8b_1ha_do)

MannwhitneyuResult(statistic=291.5, pvalue=0.34349243083925296)

In [6]:
st.mannwhitneyu(_8a_gru_do, _8b_gru_do)

MannwhitneyuResult(statistic=276.0, pvalue=0.23169571569915576)

Такие же результаты получили и для двух остальных видов бросков.<br>
Можно сделать вывод, что подготовка учеников до проведения методики в двух классах примерна одинакова.
<br><br>
Выясним, как повлияло обучение по методике на результат. Для этого вычтем из результатов после результаты до проведения методики в обоих классах и сравним их:  

In [14]:
_8a_niz_raz =[_8a_niz_ps[_] - _8a_niz_do[_] for _ in range(len(_8a_niz_ps))]
_8b_niz_raz =[_8b_niz_ps[_] - _8b_niz_do[_] for _ in range(len(_8b_niz_ps))]

In [15]:
st.mannwhitneyu(_8a_niz_raz, _8b_niz_raz)

MannwhitneyuResult(statistic=269.0, pvalue=0.1819144758639507)

**pvalue = 0.1819... больше 0.05**. Для бросков снизу проведённая методика не повлияла на результат.

In [16]:
_8a_1ha_raz =[_8a_1ha_ps[_] - _8a_1ha_do[_] for _ in range(len(_8a_1ha_ps))]
_8b_1ha_raz =[_8b_1ha_ps[_] - _8b_1ha_do[_] for _ in range(len(_8b_1ha_ps))]

In [17]:
st.mannwhitneyu(_8a_1ha_raz, _8b_1ha_raz)

MannwhitneyuResult(statistic=176.5, pvalue=0.0023602242820629183)

In [18]:
_8a_gru_raz =[_8a_gru_ps[_] - _8a_gru_do[_] for _ in range(len(_8a_gru_ps))]
_8b_gru_raz =[_8b_gru_ps[_] - _8b_gru_do[_] for _ in range(len(_8b_gru_ps))]

In [19]:
st.mannwhitneyu(_8a_gru_raz, _8b_gru_raz)

MannwhitneyuResult(statistic=141.5, pvalue=0.00011809641955195683)

**pvalue = 0.0023; pvalue = 0.0001... существенно меньше 0.05**,<br> а значит, что для бросков одной рукой и для бросков от груди проведённая методика серьёзно **улучшила результаты**.